interfaces/portal/src/app/services/auth.service.ts

```diff
@@ -202,4 +207,15 @@ export class AuthService {
     }
     return false;
   }
+
+  public async refreshUser() {
+    const currentUser = await this.queryClient.fetchQuery(
+      this.userApiService.getCurrent()(),
+    );
+    setUserInLocalStorage({
+      // we merge the existing user (to keep e.g. the token expiration), with the refreshed data from the backend
```

Comment:
Maybe use lodash's merge here? `permissions` is a nested object of user and deeper merge is more future proof.